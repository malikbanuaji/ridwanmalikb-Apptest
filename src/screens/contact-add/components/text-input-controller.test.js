import * as React from 'react';

import {render} from '@testing-library/react-native';
import TextInputController from './text-input-controller';

jest.mock('./form-text-input', () => 'FormTextInput');

test('should render properly', () => {
  const props = {
    control: 'control',
    errors: {firstName: 'firstName'},
    name: 'firstName',
    rules: {},
    label: 'label',
    placeholder: 'placeholder',
    keyboardType: undefined,
  };
  const component = render(<TextInputController {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render properly with error', () => {
  const props = {
    control: 'control',
    errors: {firstName: 'firstName'},
    name: 'lastName',
    rules: {},
    label: 'label',
    placeholder: 'placeholder',
    keyboardType: undefined,
  };
  const component = render(<TextInputController {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
