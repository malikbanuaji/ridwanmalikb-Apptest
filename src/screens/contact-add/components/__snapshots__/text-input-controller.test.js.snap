// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<View
  style={
    Object {
      "marginTop": 15,
    }
  }
>
  <Controller
    control="control"
    name="firstName"
    rules={Object {}}
  >
    <FormTextInput
      isError="firstName"
      label="label"
      onChangeText={[MockFunction]}
      placeholder="placeholder"
      value={[MockFunction]}
    />
  </Controller>
  <View
    style={
      Object {
        "marginTop": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ff3800",
          "fontSize": 14,
          "letterSpacing": 0.1,
        }
      }
    />
  </View>
</View>
`;

exports[`should render properly with error 1`] = `
<View
  style={
    Object {
      "marginTop": 15,
    }
  }
>
  <Controller
    control="control"
    name="lastName"
    rules={Object {}}
  >
    <FormTextInput
      label="label"
      onChangeText={[MockFunction]}
      placeholder="placeholder"
      value={[MockFunction]}
    />
  </Controller>
  <View
    style={
      Object {
        "marginTop": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ff3800",
          "fontSize": 14,
          "letterSpacing": 0.1,
        }
      }
    />
  </View>
</View>
`;
